rust/src/push/base_rules.rs

```diff
@@ -274,6 +274,156 @@ pub const BASE_APPEND_UNDERRIDE_RULES: &[PushRule] = &[
         default: true,
         default_enabled: true,
     },
+    PushRule {
+        rule_id: Cow::Borrowed(
+            "global/underride/.org.matrix.msc3933.rule.extensible.encrypted_room_one_to_one",
+        ),
+        priority_class: 1,
+        conditions: Cow::Borrowed(&[
+            Condition::Known(KnownCondition::EventMatch(EventMatchCondition {
```

Comment:
Regarding conditions order, does it matter (I am a bit new to push rules) ?
They are not in the same order as in the MSC. Did you tweak the order for performance reason for example ?

Reply:
I don't think it matters, as the whole array is effectively a giant `if` statement: if any of the conditions fail, the push rule's execution fails, so it shouldn't matter if the condition "fails" first or last.

Reply:
Ok so just pointing that perhaps we can improve perfs by trying to order the conditions from most likely to fail to least likely.
The `likely` part will however change quite drastically between right after deploy (almost no extensible compatible rooms) and a year later when a lot of rooms would have migrate/be created with the new room version.
And this is probably over optimization I guess, so I would say let's keep it like that and revisit if it happens to be a hot code path later on.